docs/tutorials/tabular_prediction/tabular-indepth.md

```diff
@@ -46,159 +57,277 @@ gbm_options = { # specifies non-default hyperparameter values for lightGBM gradi
+
+One option is to specify more lightweight `presets`:
+
+```{.python .input}
+presets = ['good_quality_faster_inference_only_refit', 'optimize_for_deployment']
+predictor_light = task.fit(train_data=train_data, label=label_column, eval_metric=metric,
+                           presets=presets, time_limits=30)
+```
+
+Another option is to specify more lightweight hyperparameters:
+
+```{.python .input}
+predictor_light = task.fit(train_data=train_data, label=label_column, eval_metric=metric,
+                           hyperparameters='very_light', time_limits=30)
+```
+
+Here you can set `hyperparameters` to 'light','very_light', 'toy' to obtain progressively smaller (but less accurate) models and predictors. Advanced users may instead try manually specifying particular models' hyperparameters in order to make them faster/smaller.
+
+Finally, you may also exclude specific unwieldy models from being trained at all. Below we exclude models that tend to be slower (K Nearest Neighbors, Neural Network, models with custom larger-than-default  hyperparameters):
+
+```{.python .input}
+excluded_model_types = ['KNN','NN','custom']
+predictor_light = task.fit(train_data=train_data, label=label_column, eval_metric=metric,
+                           excluded_model_types=excluded_model_types, time_limits=30)
+```
+
+
+## If you encounter memory issues
+
+To reduce memory usage, you may try each of the following strategies individually or combinations of them (these may harm accuracy):
+
+- In `fit()`, set `num_bagging_sets = 1` (can also try values greater than 1 to harm accuracy less).
+
+- In `fit()`, set `excluded_model_types = ['KNN','XT','RF']` (or some subset of these models).
+
+- Try different `presets` in `fit()`.
+
+- In `fit()`, set `hyperparameters = ‘light’` or `hyperparameters = 'very_light'`.
+
+- Text fields in your table require substantial memory for N-gram featurization. To mitigate this in `fit()`, you can either: (1) add `'ignore_text'` to your `presets` list (to ignore text features), or specify:
+
+`feature_generator = autogluon.utils.tabular.features.generators.auto_ml_pipeline.AutoMLPipelineFeatureGenerator(vectorizer=CountVectorizer(min_df=30, ngram_range=(1, 3), max_features=MAX_NGRAM, dtype=np.uint8))`
+
+where `MAX_NGRAM = 1000` say (try various values under 10000) and [CountVectorizer](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.CountVectorizer.html) is imported from `sklearn.feature_extraction.text` (to reduce the number of N-grams used to represent each text field).
 
-- Specify the `auto_stack` argument, which allows AutoGluon to automatically construct model ensembles based on multi-layer stack ensembling with repeated bagging, and will greatly improve the resulting predictions if granted sufficient training time.
+- To reduce memory required in inference, call `predictor.refit_full()` and use one of the refit-full models for prediction.
```

Comment:
This is not technically true if the models are not all being persisted.

Reply:
replaced with:

- If models have been previously persisted in memory, bagging was used in `fit()`, and inference-speed is a concern: call `predictor.refit_full()` and use one of the refit-full models for prediction (ensure this is the only model persisted in memory).